mkosi/distributions/gentoo.py

```diff
@@ -481,83 +396,30 @@ def invoke_emerge(
         else:
             cmd = ["/usr/bin/emerge", *pkgs, *self.emerge_default_opts, *opts, *actions]
 
-            MkosiPrinter.print_step("Invoking emerge(1) inside stage3")
-            run_workspace_command(
-                state,
-                cmd,
-                network=True,
-                env=self.emerge_vars,
-                nspawn_params=self.DEFAULT_NSPAWN_PARAMS,
-            )
-
-    def baselayout(self, state: MkosiState) -> None:
-        # TOTHINK: sticky bizness when when image profile != host profile
-        # REMOVE: once upstream has moved this to stable releases of baselaouy
-        # https://gitweb.gentoo.org/proj/baselayout.git/commit/?id=57c250e24c70f8f9581860654cdec0d049345292
-        self.invoke_emerge(state, inside_stage3=False,
-                           opts=["--nodeps"],
-                           pkgs=["=sys-apps/baselayout-9999"])
-
-    def update_stage3(self, state: MkosiState) -> None:
-        # exclude baselayout, it expects /sys/.keep but nspawn mounts host's
-        # /sys for us without the .keep file.
-        opts = self.EMERGE_UPDATE_OPTS + ["--exclude",
-                                          "sys-apps/baselayout"]
-        self.invoke_emerge(state, pkgs=self.pkgs_sys, opts=opts)
-
-        # FIXME?: without this we get the following
-        # Synchronizing state of sshd.service with SysV service script with /lib/systemd/systemd-sysv-install.
-        # Executing: /lib/systemd/systemd-sysv-install --root=/var/tmp/mkosi-2b6snh_u/root enable sshd
-        # chroot: failed to run command ‘/usr/sbin/update-rc.d’: No such file or directory
-        state.root.joinpath("etc/init.d/sshd").unlink()
-
-        # "build" USE flag can go now, next time users do an update they will
-        # safely merge baselayout without that flag and it should be fine at
-        # that point.
-        self.baselayout_use.unlink()
-
-    def depclean(self, state: MkosiState) -> None:
-        self.invoke_emerge(state, actions=["--depclean"])
+            MkosiPrinter.print_step("Invoking emerge(1) inside stage3"
+                                    f"{self.root}")
+            run_workspace_command(self.state, cmd, network=True, env=self.emerge_vars,
+                                  nspawn_params=self.DEFAULT_NSPAWN_PARAMS,
+                                  check=check)
```

Comment:
Why does this have to run inside the image? We're trying to move all the package manager stuff to run outside of the image instead of inside the image

Reply:
we ran into number of problems using host's `portage` in the past. only initial `sync_portage_tree()` uses host's `portage` and i'm tempted to even remove that.

also we still do make some changes related to `systemd` and we also install kernel related packages for bootable cases, see `pkgs['boot']` especially under `update_stage3()`.
also any extra packages asked by user also needs this, see `merge_user_pkgs()`.

Reply:
I would prefer if we use host's portage if possible. We do the same for all other distributions. But we can leave that for another PR since this one is already been open for a long time.

Reply:
i started out using host's `portage` and its python interface in face, which turned out to be horrible. even @poettering mentioned that we probably do the same for `dnf`...

yeah changing to host's would be a big big change with lots of test. although with usermerge in place we're dropping lots of code and complexity. i will have a go at it once day job gets a bit lighter.

Reply:
keeping this unresolved for future ref.